add docs for or use and how to extend django drip (#93)
rootstrap/django-drip-campaigns: docs/source/content/extending.rst

Extending Django Drip
=====================

> Django Drip provides a way for you to extend all the functionality
> Trough abstract classes.

Reviewer: ```suggestion
Django Drip provides a way for you to extend all the functionality trough abstract classes.
```

Reviewer 2: There's also a typo there. trough -> through

Author: good catch!

These classes provide all the functions required to make a drip work, but they don't provide any data column.

> Beware tough these classes use the attributes from the concrete drip casses shown in the rest of the docs.

Reviewer 3: `tough` means `duro` in spanish.
What do you want to write here?

Reviewer 3: I think you have a typo in `casses`.

> Beware tough these classes use the attributes from the concrete drip casses shown in the rest of the docs.

Reviewer: ```suggestion
Beware though, these classes use the attributes from the concrete drip classes shown in the rest of the docs.
```

> For you to extend them you need to overwite the implementations with your own.

Reviewer 3: ```suggestion
For you to extend them you need to overwite the implementations on your own.
```

> Since this classes handle themselves mainly trough related models, as long as you maintain the inteface you can extend in any way you want.

Reviewer 3: ```suggestion
Since this classes handle themselves mainly through related models, as long as you maintain the inteface you can extend in any way you want.
```

`AbstractDrip`

> This provides the data model for the drip, here you can customize the data model of the message itself
> it has a `drip` property with the actual logic of sending the message.

Reviewer 3: ```suggestion
This provides the data model for the drip, here you can customize the data model of the message itself.
It has a `drip` property with the actual logic of sending the message.
```

> 

Reviewer: ```suggestion
```

`AbstractSentDrip`

This provides the data model for the log of sent drips.

> If you want to customize what's being saved on the database after a message is sent

Reviewer 3: ```suggestion
If you want to customize what's being saved on the database after a message is sent.
```

> Defines a relationship with the User model of your app, that you can access trough `user.sent_drips`
> Defines a relationship with the Drip model, you can access that relationship trough `drip.sent_drips`

Reviewer 3: ```suggestion
Defines a relationship with the User model of your app, that you can access through `user.sent_drips`
Defines a relationship with the Drip model, you can access that relationship through `drip.sent_drips`
```

`AbstractQuerySetRule`

This provides the query rules applied for sending the drips.

> Defines a relationship with the Drip model you can access that relationship trough `drip.queryset_rules`

Reviewer 3: ```suggestion
Defines a relationship with the Drip model. You can access to that relationship through `drip.queryset_rules`.
```
